tgg/research/gnomad_tgps.py

```diff
@@ -0,0 +1,296 @@
+    # Get pangenome sample IDs.
+    pangenome_ids = get_pangenome_ids(pangenome_ids_path)
+    logger.info(
+        "The total number of samples in pangenome is %i", hl.eval(hl.len(pangenome_ids))
+    )
+
+    # Filter mt to contain gnomAD samples not in pangenome.
+    mt = mt.filter_cols(~pangenome_ids.contains(mt.s))
+
+    # Annotate mt with allele frequencies and filter out rare SNVs (AF lower than 1%).
+    # Also filter out sites where no samples had a variant call.
+    mt = mt.annotate_rows(freq=gnomad_freq_ht[mt.row_key].freq)
+    mt = mt.filter_rows(mt.freq.AF[0] > 0.01)
+    mt = mt.filter_rows(hl.agg.any(mt.GT.is_non_ref()))
+
+    # Read the HGDP+1KG subset matrix table and split multi-allelics.
+    pg_mt = hl.read_matrix_table(HGDP_SUBSET_MT)
+    pg_mt = hl.experimental.sparse_split_multi(pg_mt)
+    pg_mt = pg_mt.filter_rows(hl.len(pg_mt.alleles) == 1)
+
+    # Filter out sites where no samples had a variant call.
+    pg_mt = pg_mt.filter_cols(pangenome_ids.contains(pg_mt.s))
+    pg_mt = pg_mt.filter_rows(hl.agg.any(pg_mt.GT.is_non_ref()))
+
+    # Log the numbers of gnomAD samples not in and in the pangenome reference.
+    num_in_samples = pg_mt.count_cols()
+    num_out_of_samples = mt.count_cols()
+    logger.info(
+        "The number of samples both in pangenome and gnomAD v3 is: %i", num_in_samples
+    )
+    logger.info("The number of samples only in gnomAD v3 is: %i", num_out_of_samples)
+
+    # Checkpoint the rows from the pangenome MT if no checkpoint file exists.
+    pg_ht = pg_mt.rows()
+    if not hfs.is_file(PANGENOME_CHECKPOINT_FILEPATH):
+        pg_ht.checkpoint(PANGENOME_CHECKPOINT_FILEPATH)
+
+    # Annotate the gnomAD matrix table to count the number of minor alleles carried
+    # by each participant.
+    mt = mt.annotate_cols(total_var_per_sample=hl.agg.count_where(mt.GT.is_non_ref()))
+
+    # Filter out SNVs that are in the reference group.
+    pg_ht = hl.read_table(
+        PANGENOME_CHECKPOINT_FILEPATH, _n_partitions=NUM_OF_PARTITIONS
+    )
+    logger.info(
+        "The number of variants for the pangenome data set is %i", pg_ht.count()
+    )
+    mt_filtered = mt.anti_join_rows(pg_ht)
+
+    # Annotate the SNV filtered gnomAD dataset to count the number of minor alleles
+    # carried by each participant.
+    mt_filtered = mt_filtered.annotate_cols(
+        n_var_per_sample=hl.agg.count_where(mt_filtered.GT.is_non_ref())
+    )
+
+    # Keep only the columns and annotate with population data, both continental and
+    # subcontinental ancestry.
+    ht = mt_filtered.cols()
+    ht = ht.annotate(population=meta_ht[ht.key].population_inference.pop)
+
+    # Checkpoint the ht to perform the computation if no checkpoint file exists.
+    if not hfs.is_file(NOT_IN_PANGENOME_CHECKPOINT_FILEPATH):
+        ht.checkpoint(NOT_IN_PANGENOME_CHECKPOINT_FILEPATH)
+
+    # Read the computed ht.
+    ht = hl.read_table(
+        NOT_IN_PANGENOME_CHECKPOINT_FILEPATH, _n_partitions=NUM_OF_PARTITIONS
```

Comment:
what are the number of partitions of this HT anyway? maybe you can revert to a checkpoint with
```
ht = mt.rows()
...
ht = ht.naive_coalesce(NUM_OF_PARTITIONS)
ht = ht.checkpoint(...)
```

Reply:
same comment here about not needing this read if you set the HT equal to the checkpoint